src/main/scala/mesosphere/marathon/storage/migration/Helper.scala

```diff
@@ -33,3 +46,62 @@ trait MaybeStore {
   def maybeStore(persistenceStore: PersistenceStore[_, _, _]): Option[ZkPersistenceStore] = findZkStore(persistenceStore)
 
 }
+
+/**
+  * Provides a common pattern to migrate instances. This makes testing migrations easier.
+  */
+trait InstanceMigration extends MaybeStore with StrictLogging {
+
+  /**
+    * This function traverses all instances in ZK and passes them through the migration flow before saving the updates.
+    */
+  def migrateInstances(instanceRepository: InstanceRepository, persistenceStore: PersistenceStore[_, _, _])(implicit ctx: ExecutionContext, mat: Materializer): Future[Done] = {
```

Comment:
What would you think if we made this an object, instead, and pass in `migrationFlow`? This seems to be an unnecessary use of inheritence

Reply:
Hm, I've made the change but it introduces some spots for errors. Eg one has to pass the right flow. I don't quite understand why we bu the migrations in a class *and* an object. Do you know the reasoning behind it?